### ISSUE TITLE
compiler find/add cannot find compilers in /bin

### ISSUE THREAD

**Issue by stephenmsachs** [LLNL/spack#46574]
### Steps to reproduce

On rocky9 `/bin` comes before`'/usr/bin` in root's PATH while `/bin` is a symbolic link to thhe latter. Spack will only check for compilers in the first of the two paths and will fail since `executable_prefix('/bin')` returns an empty string.
```
PATH=/bin/:usr/bin:/path/to/spack spack compiler find
```

### Error message

```
==> [2024-09-25-11:19:00.290111, 1086129] '/bin/gcc' '-dumpfullversion'
==> [2024-09-25-11:19:00.292175, 1086129] '/bin/g++' '-dumpfullversion'
checking path
/bin
==> [2024-09-25-11:19:00.294482, 1086129] no bin/ dir found in /bin. Cannot add it as a Spack package
```

### Information on your system

* **Spack:** 0.23.0.dev0 (6692eb7b6fe625bbddab2ba1a7b94483b5c086b1)
* **Python:** 3.9.18
* **Platform:** linux-rocky9-zen4

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by upsj** [comment]
This has hit me repeatedly as well, `spack compiler find /usr` is a workaround if `/bin` is symlinked to `/usr/bin`

**Comment by haampie** [comment]
Closed by #46957
